packages/sdk/src/oracle/store/write.ts

```diff
@@ -68,19 +69,18 @@ export class OptimisticOracle {
   address(address: string): void {
     this.state.address = address;
   }
-  request(inputRequest: state.Inputs["request"], request: state.Request): void {
-    const id = [
-      inputRequest.requester,
-      inputRequest.identifier,
-      inputRequest.timestamp,
-      inputRequest.ancillaryData,
-    ].join("!");
+  request(inputRequest: state.InputRequest, request: state.Request): void {
+    const id = requestId(inputRequest);
     if (!this.state.requests) this.state.requests = {};
     this.state.requests[id] = request;
   }
   defaultLiveness(defaultLiveness: ethersTypes.BigNumber): void {
     this.state.defaultLiveness = defaultLiveness;
   }
+  event(event: state.OptimisticOracleEvent): void {
+    if (!this.state.events) this.state.events = [];
+    insertOrderedAscending(this.state.events, event, eventKey);
```

Comment:
this makes sure the events maintain order in array. we basically will cache a history of all known events